### PR TITLE
Add support chat

### LINKED ISSUE #863: Add support chat
We need to add a link to support CRM users that integrates with a support provider (e.g. Intercom, Front...)

This will be mostly useful for our cloud version on app.twenty.com ; not sure if it can be of any use for people who self-host and do things on their sides. But still let's try to keep an open design if we can.

For now we will support Front as our first driver. 
Here is the SDK reference: https://dev.frontapp.com/docs/chat-sdk-reference

This isn't a purely frontend ticket because:
- The config (ex API) needs to come from the clientConfig endpoint
- We need to authenticate users through HMAC

# Backend
SUPPORT_DRIVER=front 
SUPPORT_HMAC_KEY=...
SUPPORT_FRONTEND_KEY=...

# Frontend
Here is the code to insert ont the frontend with key we will being using in production, just for the record (don't include it in the code):
```
<script src="https://chat-assets.frontapp.com/v1/chat.bundle.js"></script>
<script>
  window.FrontChat('init', {chatId: '43f23c78d5edf5f821204b8b0f27b56d', useDefaultLauncher: false});
</script>
```
Note we should probably set the defaultLauncher to false since there is a custom launcher on bottom. And make sure it opens fully when we click on that.

In terms of fields we want to pass. Let's pass email/name/hash and not implement any custom fields (we could build a plugin in Front for that)

### PR DESCRIPTION
**Pull request by gitstart-twenty**
Fixes #863

Demo: 

https://github.com/twentyhq/twenty/assets/140154534/a1237dc6-8ea3-4562-ac23-dc7c019106d2

### CODE REVIEW

**Comment by FelixMalfait** [comment]
Great! Thanks!

Could you please update the style to exactly match the Figma prototype, including hover behavior? https://www.figma.com/proto/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=5910-77816&t=zWQLlwzjzomOAUIo-0&scaling=min-zoom&page-id=478%3A19796&starting-point-node-id=1406%3A44830

Here is the Figma in dev mode:
https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=5910-77816&mode=design&t=zWQLlwzjzomOAUIo-0

Also, let's not display the button if the driver hasn't been set properly. Here in dev mode I can see the button even if the .env variable hasn't been set 

Finally, I couldn't test the scroll behavior on the left menu, maybe add something like overflow:auto to make the menu scrollable? (Please test it doesn't have any other unattended side effect)

File: front/src/modules/ui/navbar/components/SupportChat.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    variables: { email: user?.email },
+  });
+
+  useEffect(() => {
+    if (!loading && data?.clientConfig) {
+      setSupportChatConfig(data?.clientConfig.supportChat);
+    }
+  }, [data, loading, setSupportChatConfig]);
+
+  useEffect(() => {
```
Comment (charlesBochet) [inline front/src/modules/ui/navbar/components/SupportChat.tsx:77]: I feel these 3 useEffects are not necessary. Is it possible to refactor this code in a mode "sequential way". We should only have 1 useEffect based on if the config is loaded to setup front (insert src script, init with defaultlauncher: false) and one on the currentuser being loaded to run the init with user params call. This looks over complicated but I might be missing some details about the way the front plugin is setup

**Review by charlesBochet — COMMENTED** [review]
Thank you! I have recommended a few changes

File: front/src/modules/ui/navbar/components/SupportChat.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  const user = useRecoilValue(currentUserState);
+  const [isFrontChatLoaded, setIsFrontChatLoaded] = useState(false);
+
+  const { data, loading } = useGetClientConfigQuery({
+    variables: { email: user?.email },
+  });
+
+  useEffect(() => {
+    if (!loading && data?.clientConfig) {
+      setSupportChatConfig(data?.clientConfig.supportChat);
```
[the remark this answers is not in the document]
Comment (gitstart-twenty) [reply front/src/modules/ui/navbar/components/SupportChat.tsx:89]: Definitely, I somehow missed this that we always fetch it 😅

File: front/src/modules/ui/navbar/components/SupportChat.tsx (hunk 1 above)
Comment (gitstart-twenty) [reply front/src/modules/ui/navbar/components/SupportChat.tsx:76]: Definitely needs to be simplified

File: front/src/modules/ui/navbar/components/SupportChat.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+const StyledQuestionMark = styled.div`
+  align-items: center;
+  border-radius: 50%;
+  border-style: solid;
+  border-width: ${({ theme }) => theme.spacing(0.25)};
+  display: flex;
+  height: ${({ theme }) => theme.spacing(3.5)};
+  justify-content: center;
+  margin-right: ${({ theme }) => theme.spacing(1)};
+  width: ${({ theme }) => theme.spacing(3.5)};
```
Comment (charlesBochet) [inline front/src/modules/ui/navbar/components/SupportChat.tsx:26]: let's not use spacing for width and height :)

File: front/src/modules/ui/navbar/components/SupportChat.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  ButtonVariant,
+} from '@/ui/button/components/Button';
+
+const StyledButtonContainer = styled.div`
+  display: flex;
+`;
+
+const StyledQuestionMark = styled.div`
+  align-items: center;
+  border-radius: 50%;
```
Comment (charlesBochet) [inline front/src/modules/ui/navbar/components/SupportChat.tsx:19]: you can use theme.border.radius.rounded

**Review by charlesBochet — APPROVED** [review]
LGTM with minor comments. Merging it

**Pull request merged**